deltacat/tests/compactor/test_compaction_session.py

```diff
@@ -0,0 +1,406 @@
+        source_table_version,
+        primary_key_column_names=list(primary_keys),
+        sort_keys=sort_keys,
+        partition_keys=partition_keys,
+        supported_content_types=[ContentType.PARQUET],
+        **ds_mock_kwargs,
+    )
+    source_table_stream: Stream = ds.get_stream(
+        namespace=source_namespace,
+        table_name=source_table_name,
+        table_version=source_table_version,
+        **ds_mock_kwargs,
+    )
+    test_table: pa.Table = pa.Table.from_arrays(arrow_arrays, names=column_names)
+    staged_partition: Partition = ds.stage_partition(
+        source_table_stream, partition_values, **ds_mock_kwargs
+    )
+    ds.commit_delta(
+        ds.stage_delta(test_table, staged_partition, **ds_mock_kwargs), **ds_mock_kwargs
+    )
+    ds.commit_partition(staged_partition, **ds_mock_kwargs)
+    # create the destination table
+    ds.create_namespace(destination_namespace, {}, **ds_mock_kwargs)
+    ds.create_table_version(
+        destination_namespace,
+        destination_table_name,
+        destination_table_version,
+        primary_key_column_names=list(primary_keys),
+        sort_keys=sort_keys,
+        partition_keys=partition_keys,
+        supported_content_types=[ContentType.PARQUET],
+        **ds_mock_kwargs,
+    )
+    destination_table_stream: Stream = ds.get_stream(
+        namespace=destination_namespace,
+        table_name=destination_table_name,
+        table_version=destination_table_version,
+        **ds_mock_kwargs,
+    )
+    ray.shutdown()
+    ray.init(local_mode=True)
+    assert ray.is_initialized()
+    source_table_version = ds.get_table_version(
+        TEST_SOURCE_NAMESPACE,
+        TEST_SOURCE_TABLE_NAME,
+        TEST_SOURCE_TABLE_VERSION,
+        **ds_mock_kwargs,
+    )
+    source_partition = ds.get_partition(
+        ds.get_stream(
+            namespace=TEST_SOURCE_NAMESPACE,
+            table_name=TEST_SOURCE_TABLE_NAME,
+            table_version=TEST_SOURCE_TABLE_VERSION,
+            **ds_mock_kwargs,
+        ).locator,
+        partition_values,
+        **ds_mock_kwargs,
+    )
+    destination_partition_locator = PartitionLocator.of(
+        destination_table_stream.locator,
+        partition_values,
+        None,
+    )
+    num_workers, worker_instance_cpu = 1, 1
+    total_cpus = num_workers * worker_instance_cpu
+    pgm = PlacementGroupManager(1, total_cpus, worker_instance_cpu).pgs[0]
+    compact_partition_params: Dict[str, Any] = {
+        "compaction_artifact_s3_bucket": TEST_S3_RCF_BUCKET_NAME,
+        "compacted_file_content_type": ContentType.PARQUET,
+        "dd_max_parallelism_ratio": 1.0,
+        "deltacat_storage": ds,
+        "destination_partition_locator": destination_partition_locator,
+        "hash_bucket_count": None,
+        "last_stream_position_to_compact": source_partition.stream_position,
+        "list_deltas_kwargs": {**ds_mock_kwargs, **{"equivalent_table_types": []}},
+        "pg_config": pgm,
```

Comment:
can we parametrize create_placement_group boolean? Its not mandatory to create pg in v2,.

Reply:
Done